misk/src/main/kotlin/misk/web/exceptions/IllegalArgumentExceptionMapper.kt

```diff
@@ -0,0 +1,23 @@
+package misk.web.exceptions
+
+import misk.exceptions.StatusCode
+import misk.web.Response
+import misk.web.ResponseBody
+import misk.web.marshal.StringResponseBody
+import misk.web.mediatype.MediaTypes
+import okhttp3.Headers
+
+/** Maps [IllegalArgumentException]s into BAD_REQUEST responses */
+internal class IllegalArgumentExceptionMapper : ExceptionMapper<IllegalArgumentException> {
```

Comment:
Yeah, I think this is not something we should do by default.

We _may_ want a nice API to check an action’s arguments and throw a BadRequestException

Reply:
Hmm, ok, reason I sortof like this is that you can write library code that doesn't need to be aware of the fact that it is running in an action. But I can support removing this and adding the equivalent of ``check() {}`` to raise a ``BadRequestException`` instead of ``IllegalArgumentException``.